src/detection_win.cpp

```diff
@@ -364,104 +395,111 @@ void ExtractDeviceInfo(HDEVINFO hDevInfo, SP_DEVINFO_DATA* pspDevInfoData, TCHAR
 	resultItem->deviceAddress = dummy++;
 
 	// device found
-	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_FRIENDLYNAME, &DataT, (PBYTE)buf, buffSize, &nSize)) {
+	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_FRIENDLYNAME, &DataT, (PBYTE)buf, MAX_PATH, &nSize)) {
 		resultItem->deviceName = buf;
 	}
-	else if ( DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_DEVICEDESC, &DataT, (PBYTE)buf, buffSize, &nSize))
+	else if ( DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_DEVICEDESC, &DataT, (PBYTE)buf, MAX_PATH, &nSize))
 	{
 		resultItem->deviceName = buf;
 	}
-	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_MFG, &DataT, (PBYTE)buf, buffSize, &nSize)) {
+	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_MFG, &DataT, (PBYTE)buf, MAX_PATH, &nSize)) {
 		resultItem->manufacturer = buf;
 	}
-	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_HARDWAREID, &DataT, (PBYTE)buf, buffSize, &nSize)) {
-		// Use this to extract VID / PID
-		extractVidPid(buf, resultItem);
-	}
+	ExtractDetails(devicePath, resultItem);
 }
 
+void HandleDeviceConnect(PDEV_BROADCAST_DEVICEINTERFACE pDevInf) {
+	HDEVINFO hDevInfo = DllSetupDiGetClassDevs(&GUID_DEVINTERFACE_USB_DEVICE, NULL, NULL, DIGCF_PRESENT | DIGCF_DEVICEINTERFACE);
+	if (INVALID_HANDLE_VALUE == hDevInfo) {
```

Comment:
nit, prefer `hDevInfo == INVALID_HANDLE_VALUE` (subject before comparison constant)

(same with other if-statements in these changes)

Reply:
After reviewing the file again, I believe changing all the comparisons to be subject before constant would be trivial and can add that to the patch update.